### PR TITLE
Fix: can't convert Transform -> Transform2D

### LINKED ISSUE #38528: Can't convert Transform to Transform2D
**Godot version:** 3.2.1
**OS/device including version:** 4.19.97-gentoo
**Issue description:**
I need to make projection to XY plane.
I try to convert Transform to Transform2d as written in the documentation https://docs.godotengine.org/en/3.2/classes/class_transform2d.html#class-transform2d-method-transform2d
But i get error: Invalid call. Nonexistent function 'get_rotation' in base 'Nill'.
**Steps to reproduce:**

**Minimal reproduction project:**
    in script inherited from Spatial:
    var s_trans = get_transform() # Transform
    var t = Transform2D(s_trans)
    t.get_rotation()

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #38528

### CODE REVIEW

File: core/variant_call.cpp [hunk 1]
```diff
@@ -1360,6 +1360,8 @@ Variant Variant::construct(const Variant::Type p_type, const Variant **p_args, i
 			case BASIS: return (Basis(p_args[0]->operator Basis()));
 			case TRANSFORM:
 				return (Transform(p_args[0]->operator Transform()));
+			case TRANSFORM2D:
+				return (Transform2D(p_args[0]->operator Transform2D()));
```
Comment (akien-mga) [inline core/variant_call.cpp:1364]: It should come after `case VECTOR3I` to keep the same order as in `core/variant.h`.

Comment (ThakeeNathees) [reply core/variant_call.cpp]: @akien-mga  done 👍

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**